### DISCUSSION TITLE
SpikingJelly 现已加入 GitLink 编程夏令营（GLCC）

### DISCUSSION THREAD

**Discussion by ridgerchu** [fangwei123456/SpikingFlow#225] [Show and tell]
[GitLink编程夏令营（GLCC）](https://www.gitlink.org.cn/glcc)，是在CCF中国计算机学会指导下，由CCF开源发展委员会（CCF ODC）举办的面向全国高校学生的暑期编程活动。入围夏令营的同学们将获得现金支持、实习机会和开源大礼包等。
SpikingJelly项目将参与本次GLCC，可以[点击这里](https://www.gitlink.org.cn/ridger/spikingjelly)进入查看详情。因目前项目仍未正式发布，所有可选项目更新均在新链接的issue（疑修）中。学生报名将于5月26日开启，欢迎参与🎉～

**Comment by fangwei123456** [comment]
下一个版本在 [fuse_ms](https://github.com/fangwei123456/spikingjelly/tree/fuse_ms)，开发以次分支为基础，而不是目前的master分支

**Comment by fangwei123456** [comment]
https://spikingjelly.readthedocs.io/zh_CN/fuse_ms/activation_based/migrate_from_legency.html

https://spikingjelly.readthedocs.io/zh_CN/fuse_ms/activation_based/basic_concept.html

**Comment by fangwei123456** [comment]
我看到GitLink那边目前是这3个issue，有点太简单了：
![image](https://user-images.githubusercontent.com/38352109/170611645-a4f642a4-9297-4b2a-9dbc-5d4475812f3b.png)

我建议改成下面3个：

1. 所有神经元的单步/多步CuPy内核的实现，并且增加v_rest。在目前的版本中是将v_reset作为v_rest，有必要对两者进行区分了。

2. [LAVA框架](https://github.com/lava-nc/lava)适配。LAVA是intel新推出的框架，未来将支持在loihi 1/2上进行推理。若想将GPU上训练的网络迁移到神经形态芯片，则SJ框架需要通过适配LAVA框架来完成。

3. GPU计算加速。目前考虑可能的路线有两条，一条是利用脉冲的稀疏特性，使用稀疏卷积和稀疏矩阵乘法进行加速。在SJ的早期探索过稀疏矩阵乘法，但是发现只有在稀疏度极高的情况下才能加速。稀疏卷积还没有尝试过。另一条技术路线是利用 [https://github.com/NVIDIA/cutlass](https://github.com/NVIDIA/cutlass) 开发新的算子，实现混合精度计算加速。这里的混合精度不是AMP那种混合精度，而是使用fp32/fp16@int8/int4/int1的混合计算，例如fp16的weight和int1的脉冲进行卷积计算。

**Reply by fangwei123456** [reply]
这3个的难度递增，1是偏工程的，没什么难度但工程量大。3是最难的

**Reply by ridgerchu** [reply]
我们正在修改项目内容，将修改为如上三个项目。
